lightcurve/views.py

```diff
@@ -180,5 +182,27 @@ def all_lightcurve(request):
             'images': images,
         })
 
-    return render_to_response('lightcurve_list.html', {"contexts": context_list, "request_all": True},
+    return render_to_response('lightcurve_list.html', {'contexts': context_list, 'request_all': True},
             context_instance=RequestContext(request))
+
+
+def download(request, lightcurve_id):
+    file_type = request.GET.get('file_type')
+    response = HttpResponse(content_type='text/csv')
+    response['Content-Disposition'] = 'attachment; filename="LightCurve%d.%s"' % (int(lightcurve_id), file_type) # Gets TypeError without int(). %s instead of %d?
+
+    lc = LightCurve.objects.get(id=lightcurve_id)
+    analyses = ImageAnalysis.objects.filter(useruploadedimage__lightcurve=lc) \
+                                    .exclude(status=ImageAnalysis.ASTROMETRY_PENDING)
+
+    if file_type == 'csv':
+        writer = csv.writer(response)
+        writer.writerow(['Datetime', 'JD', 'Mag instrumental', 'Mag standard', 'Mag std'])
+        for analysis in analyses:
+            if analysis.annotated_point_sources != "[]":
```

Comment:
I think you are getting this error because there is a malformed value in the database.  This is my fault because I recall an issue with the test db I sent you.

Try going to http://localhost:8000/admin/ (you will have to `./manage.py createsuperuser` if you haven't already).  Then go to Image Analysis --> [Image # that is causing this error] and edit the `annotated_point_sources` field to be `[]` rather than `"[]"`.